Shop.js

```diff
@@ -0,0 +1,155 @@
+// Магазин
+
+const Product = function (name, type, cost) {
+    this.name = name;
+    this.type = type;
+    this.cost = cost;
+}
+
+const Client = function (name, surname) {
+    this.name = name;
+    this.surname = surname;
+    this.sum = 0;
+}
+
+const PurchaseHistoryItem = function (product, number = 0) {
+    this.productName = product.name;
+    this.productType = product.type;
+    this.productNumber = number;
+    this.itemCost = product.cost * number;
+
+    this.time = function () {
+        let date = new Date();
+        let hours = date.getHours() > 10 ? date.getHours().toString() : "0" + date.getHours().toString();
+        let minutes = date.getMinutes() > 10 ? date.getMinutes().toString() : "0" + date.getMinutes().toString();
+        let seconds = date.getSeconds() > 10 ? date.getSeconds().toString() : "0" + date.getSeconds().toString();
+        return `${hours}:${minutes}:${seconds}`;
+    }();
+};
+
+
+const Shop = function () {
+    this.productsList = new Map();
+    this.clientsList = [];
+    this.purchaseHistoryList = [];
+    this.totalNumber = 0;
+};
+
+Shop.prototype = {
+    addClient: function (client) {
+        this.clientsList.push(client);
+    },
+
+    deleteClient: function (index) {
+        this.clientsList.splice(index, 1);
+    },
+
+    addProduct: function (product, number = 1) {
+        if (this.productsList.has(product.name)) {
+            this.productsList.get(product.name).counter += number;
+        }
+        else {
+            let buf = {
+                product: product,
```

Comment:
Codacy has a fix for the issue: [Expected property shorthand.](https://app.codacy.com/gh/Varlyle/my-first-project/pullRequest?prid=7306920)
```suggestion
                product,
```